commonmark/src/main/java/org/commonmark/internal/DocumentParser.java

```diff
@@ -81,6 +84,27 @@ public Document parse(String input) {
         this.processInlines();
         return documentBlockParser.getBlock();
     }
+    
+    public Document parse(Reader input) throws IOException {
+        DocumentBlockParser documentBlockParser = new DocumentBlockParser();
+        activateBlockParser(documentBlockParser);
```

Comment:
If we put those two lines into the contructor, then we don't have to repeat them in both parse methods.